### PR TITLE
Vidshare Service

### PR DESCRIPTION
**Pull request by dough321**

### CODE REVIEW

**Comment by sander1** [comment]
There are some things that couldn't make this URL Service work. I'll try to add some hints to the code in the PR. A service for this site has been written and added here: https://github.com/plexinc-plugins/Services.bundle/blob/master/Contents/Service%20Sets/com.plexapp.plugins.pftv/URL/letwatch.us/ServiceCode.pys
It'll take some time before this goes live.

File: Contents/Service Sets/com.plexapp.plugins.vishare/ServiceInfo.plist [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<plist version="1.0">
+<dict>
+  <key>URL</key>
+  <dict>
+    <key>Vidshare</key>
+    <dict>
+      <key>URLPatterns</key>
+      <array>
+        <string>http://vidshare.us</string>
```
Comment (sander1) [inline Contents/Service Sets/com.plexapp.plugins.vishare/ServiceInfo.plist]: This pattern is not specific for 1 video.

File: Contents/Service Sets/com.plexapp.plugins.vishare/ServiceInfo.plist [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+		<string>http://vidshare.us/.+</string>
```
Comment (sander1) [inline Contents/Service Sets/com.plexapp.plugins.vishare/ServiceInfo.plist]: Videos on the site seem to use the `letwatch.us` domain, not `vidshare.us`, this regex will not match the real URLs.

File: Contents/Service Sets/com.plexapp.plugins.vishare/URL/VIDSHARE/ServiceCode.pys [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import string, re
```
Comment (sander1) [inline Contents/Service Sets/com.plexapp.plugins.vishare/URL/VIDSHARE/ServiceCode.pys]: Avoid importing modules where possible. Most common string functions are included in the Plex Framework. Same goes for the `re` module.

File: Contents/Service Sets/com.plexapp.plugins.vishare/URL/VIDSHARE/ServiceCode.pys [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+HTTP_HEADERS = {
```
Comment (sander1) [inline Contents/Service Sets/com.plexapp.plugins.vishare/URL/VIDSHARE/ServiceCode.pys]: This is probably not necessary, it's not even used in the code below :)

File: Contents/Service Sets/com.plexapp.plugins.vishare/URL/VIDSHARE/ServiceCode.pys [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+def NormalizeURL(url):
+
+	return url
+
+####################################################################################################
+def MetadataObjectForURL(url):
+	
+	try:
+		page_data = HTML.ElementFromURL(url)
+		keys = page_data.xpath("//script//text()")
```
Comment (sander1) [inline Contents/Service Sets/com.plexapp.plugins.vishare/URL/VIDSHARE/ServiceCode.pys]: The `script` block we want is not present on the page when you first load it. You need to do a POST (press the 'Proceed to Video' button) to get to this code.

File: Contents/Service Sets/com.plexapp.plugins.vishare/URL/VIDSHARE/ServiceCode.pys [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+		if mykey.find('|') != -1:
+			mykey = mykey.split('|');
+			mykey = mykey[len(mykey)-1]
+		
+		url0 = host + '/' + mykey + '/' + 'v.flv'
+		# http://103.43.94.69/l7z7btxjdqnhgn4vfj2bga6cqdroexc376gj3a6mmo4emsbchtortpisujva/v.flv
+	except:
+		url0 = url
+		
+	title = 'VidShare Redirect Page'
```
Comment (sander1) [inline Contents/Service Sets/com.plexapp.plugins.vishare/URL/VIDSHARE/ServiceCode.pys]: This is not a title you want to see in your Watch Later queue ;-)

**Comment by dough321** [comment]
Hi Sander
Thanks for the review and edits. How will I know when this URL service will be live ?
Thanks

**Comment by sander1** [comment]
It'll be active soon, probably within 48 hours.

**Comment by dough321** [comment]
This URL service is still not working :(. I tried below link
http://letwatch.us/xojx1ipaiot9

**Pull request closed without merging**